cpp/tests/test_abm_world.cpp

```diff
@@ -242,8 +233,22 @@ TEST(TestWorld, evolveMigration)
         auto work_id     = world.add_location(mio::abm::LocationType::Work);
         auto hospital_id = world.add_location(mio::abm::LocationType::Hospital);
 
-        auto& p1 =
-            add_test_person(world, home_id, mio::abm::AgeGroup::Age15to34, mio::abm::InfectionState::Susceptible, t);
+        ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>>
+            mock_uniform_dist;
+        EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
+            .Times(testing::AtLeast(8))
+            .WillOnce(testing::Return(0.8)) // draw random work group
+            .WillOnce(testing::Return(0.8)) // draw random school group
+            .WillOnce(testing::Return(0.8)) // draw random work hour
+            .WillOnce(testing::Return(0.8)) // draw random school hour
+            .WillOnce(testing::Return(0.8)) // draw random work group
+            .WillOnce(testing::Return(0.8)) // draw random school group
+            .WillOnce(testing::Return(0.8)) // draw random work hour
+            .WillOnce(testing::Return(0.8)) // draw random school hour
+            .WillRepeatedly(testing::Return(1.0)); // this forces p1 and p3 to recover
+
+        auto& p1 = add_test_person(world, home_id, mio::abm::AgeGroup::Age15to34,
+                                   mio::abm::InfectionState::InfectedNoSymptoms, t);
```

Comment:
I think the fact that we need to setup a mock during the setup of the test (not the actual thing to be tested) points to problems in the design. I think it would be better if there is always a way to set a specific state. This makes not just testing easier/clearer but also running very specific scenarios. Of course the ability to randomize things should exist as well, maybe always have two overloads, one with RNG, one with fixed values.

Some instances where this is a problem (maybe others):
- infections
- set up of persons (random work group)

These cases are more obvious now that you always need to pass in an RNG as an argument. Nothing to be done now, would be too big, but maybe we should take this into account in future development.

Reply:
I guess you are right, we should add that at some point soon.